src/overlays/actors/ovl_Boss_06/z_boss_06.c

```diff
@@ -20,54 +20,62 @@ void Boss06_Destroy(Actor* thisx, PlayState* play);
 void Boss06_Update(Actor* thisx, PlayState* play);
 void Boss06_Draw(Actor* thisx, PlayState* play2);
 
-void func_809F24A8(Boss06* this);
-void func_809F24C8(Boss06* this, PlayState* play);
-void func_809F2B64(Boss06* this, PlayState* play);
-void func_809F2C44(Boss06* this, PlayState* play);
-void func_809F2E14(Boss06* this, PlayState* play);
-void func_809F2E34(Boss06* this, PlayState* play);
-void func_809F2ED0(Boss06* this, PlayState* play);
-void func_809F2EE8(Boss06* this, PlayState* play);
-
-Vec3f D_809F4370[128];
-EnKnight* D_809F4970;
-s32 D_809F4974;
-s32 D_809F4978;
-s32 D_809F497C;
+void Boss06_SetupCurtainBurningCutscene(Boss06* this);
+void Boss06_CurtainBurningCutscene(Boss06* this, PlayState* play);
+void Boss06_SetupCloseCurtain(Boss06* this, PlayState* play);
+void Boss06_CloseCurtain(Boss06* this, PlayState* play);
+void Boss06_SetupCurtainClosed(Boss06* this, PlayState* play);
+void Boss06_CurtainClosed(Boss06* this, PlayState* play);
+void Boss06_SetupCurtainDestroyed(Boss06* this, PlayState* play);
+void Boss06_CurtainDestroyed(Boss06* this, PlayState* play);
+
+static Vec3f sFireEffectPositions[128];
+static EnKnight* sIgosInstance;
+static s32 sSeed1;
+static s32 sSeed2;
+static s32 sSeed3;
```

Comment:
I've renamed `sFireEffectPositions` to `sCurtainFireEffectPositions` and made that non-static. `sIgosInstance` is already non-static in En_Knight so can't be non-static in Boss_06

Reply:
Unless I'm missing something, can't both instances be prefixed to be made non-static? Like `sKnightIgosInstance` and `sCurtainIgosInstance`?

Reply:
Bumping this one; I *think* you an avoid static BSS here by prefixing the Igos instance in Boss06? I could be missing something as to why they both need the same name, though

Reply:
There's no reason why this couldn't be done, but I'd rather not rename them unless it's absolutely necessary. I think the names are better as they are now

Reply:
Apologies, I misunderstood how the bss script worked. I thought that any static bss would cause the script to fail, even if the file contained non-static bss, hence why we try to avoid it. However, this isn't true; if this file is causing bss issues, the script will still be able to detect that, since there's non-static bss in this file. Thus, there's no real reason to change the name of this variable, and I agree with your assessment; the variable names are good as they are. Plus, if this file *is* causing a lot of bss issues later, it's not like we can't just change this down the line either. Sorry again for misunderstanding things.